### PR TITLE
grant tokens only on the first add key and add keys

### PR DESCRIPTION
**Pull request by JustynaBroniszewska**

### CODE REVIEW

**Comment by alexvandesande** [comment]
preferably that state should be available to the app, so it can show the user if they will receive tokens or have received tokens for these actions: https://github.com/EthWorks/UniversalLoginSDK/issues/105

File: universal-login-relayer/lib/services/IdentityService.js [hunk 1]
```diff
@@ -33,13 +33,18 @@ class IdentityService {
   async executeSigned(contractAddress, message) {
     const contract = new ethers.Contract(contractAddress, this.abi, this.wallet);
     const addKeySighash = new Interface(Identity.interface).functions.addKey.sighash;
+    const addKeysSighash = new Interface(Identity.interface).functions.addKeys.sighash;
     if (message.to === contractAddress && message.data.slice(0, addKeySighash.length) === addKeySighash) {
```
Comment (marekkirejczyk) [inline universal-login-relayer/lib/services/IdentityService.js]: `message.data.slice(0, addKeySighash.length) === addKeySighash ` => extract function

File: universal-login-relayer/lib/services/IdentityService.js [hunk 2]
```diff
@@ -33,13 +33,18 @@ class IdentityService {
...
       const [address] = (this.codec.decode(['bytes32', 'uint256', 'uint256'], message.data.replace(addKeySighash.slice(2), '')));
       const key = utils.hexlify(utils.stripZeros(address));
       await this.authorisationService.removeRequest(contractAddress, key);
       const transaction = await contract.executeSigned(message.to, message.value, message.data, message.nonce, message.gasToken, message.gasPrice, message.gasLimit, message.signature);
       this.hooks.emit('added', key);
       return transaction;
+    } else if (message.to === contractAddress && message.data.slice(0, addKeysSighash.length) === addKeysSighash) {
```
Comment (marekkirejczyk) [inline universal-login-relayer/lib/services/IdentityService.js]: `message.data.slice(0, addKeysSighash.length) === addKeysSighash ` => extract function

File: universal-login-relayer/lib/services/IdentityService.js [hunk 3]
```diff
@@ -33,13 +33,18 @@ class IdentityService {
   async executeSigned(contractAddress, message) {
     const contract = new ethers.Contract(contractAddress, this.abi, this.wallet);
     const addKeySighash = new Interface(Identity.interface).functions.addKey.sighash;
+    const addKeysSighash = new Interface(Identity.interface).functions.addKeys.sighash;
     if (message.to === contractAddress && message.data.slice(0, addKeySighash.length) === addKeySighash) {
       const [address] = (this.codec.decode(['bytes32', 'uint256', 'uint256'], message.data.replace(addKeySighash.slice(2), '')));
       const key = utils.hexlify(utils.stripZeros(address));
       await this.authorisationService.removeRequest(contractAddress, key);
```
Comment (marekkirejczyk) [inline universal-login-relayer/lib/services/IdentityService.js]: last 3 lines -> extract function

**Pull request merged**